### PR TITLE
Modify rule S2183: LaYC format

### PR DESCRIPTION
**Pull request by gregory-paidis-sonarsource**

### CODE REVIEW

File: rules/S2183/csharp/rule.adoc [hunk 1]
```diff
@@ -1,62 +1,64 @@
...
-Note that integral number with a less than 32-bit quantity (e.g. ``++short++``, ``++ushort++``...) are implicitly converted to ``++int++`` before the shifting operation and so the rule for ``++int++``/``++uint++`` applies.
+[source,csharp]
+----
+var one =         0b0_00001;
+var thirtyThree = 0b1_00001; // Same five low-order bits, 33 mod 32 = 1
+----
+
+Note that integral number with a less than 32-bit quantity (e.g. `short`, `ushort`...) are implicitly converted to `int` before the shifting operation and so the rule for `int`/`uint` applies.
 
+If the first operand is a `long` or `ulong` (64-bit quantity), the shift count is given by the low-order six bits of the second operand. That is, the actual shift count is 0 to 63 bits.
```
Comment (sebastien-marichal) [inline rules/S2183/csharp/rule.adoc:17]:
```suggestion
If the first operand is a `long` or `ulong` (64-bit quantity), the shift count is given by the six low-order bits of the second operand. That is, the actual shift count is 0 to 63 bits.
```

File: rules/S2183/csharp/rule.adoc [hunk 2]
```diff
@@ -1,62 +1,64 @@
 == Why is this an issue?
 
-Shifting an integral number by 0 is equivalent to doing nothing but makes the code confusing for maintainers.
+https://en.wikipedia.org/wiki/Arithmetic_shift[Shifting] an integral number by 0 is equivalent to doing nothing but makes the code confusing for maintainers.
 
+Shifting an integral number by a value greater than their count of bits minus one is equivalent to shifting by the value https://en.wikipedia.org/wiki/Modulo[modulo] the bit count of the number.
 
-If the first operand is an ``++int++`` or ``++uint++`` (32-bit quantity), the shift count is given by the low-order five bits of the second operand. That is, the actual shift count is 0 to 31 bits.
+If the first operand is an `int` or `uint` (32-bit quantity), the shift count is given by the five low-order bits of the second operand. That is, the actual shift count is 0 to 31 bits. This means that numbers having the same five low-order bits are treated the same by the shift-operators.
```
Comment (sebastien-marichal) [inline rules/S2183/csharp/rule.adoc:7]: I have to admit, it took me several tries to fully understand what this is about. Especially the part about the low-order bits.
Also, I am not sure it is relevant to talk about Modulo here.

It is a bit too technical for a beginner to understand but in the meantime, it has to be somehow.

File: rules/S2183/csharp/rule.adoc [hunk 3]
```diff
@@ -1,62 +1,64 @@
...
-    long longShift2 = lg << 64; // Noncompliant, this is equivalent to shifting by 1
-}
+=== Code examples
+
+==== Noncompliant code example
+
+[source,csharp,diff-id=1,diff-type=noncompliant]
 ----
+short s = 1;
+short shortShift1 = (short)(s << 0); // Noncompliant: The value does not change
```
Comment (sebastien-marichal) [inline rules/S2183/csharp/rule.adoc:30]:
```suggestion
short shortShift1 = (short)(s << 0); // Noncompliant: the value does not change
```

**Review by sebastien-marichal — CHANGES_REQUESTED** [review]
I left typo comments.
Also, I think the description is a bit too technical in my opinion but I understand it is difficult to be simpler.

**Comment by gregory-paidis-sonarsource** [comment]
> I left typo comments. Also, I think the description is a bit too technical in my opinion but I understand it is difficult to be simpler.

Well said. I gave it a try to make it more understandable, tell me what you think.

**Review by sebastien-marichal — APPROVED** [review]
The description seems to me way simpler than before 👏 
I left a suggestion
LGTM!

File: rules/S2183/csharp/rule.adoc [hunk 4]
```diff
@@ -1,20 +1,32 @@
...
 
-https://en.wikipedia.org/wiki/Arithmetic_shift[Shifting] an integral number by 0 is equivalent to doing nothing but makes the code confusing for maintainers.
+The https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/bitwise-and-shift-operators#left-shift-operator-[shifting] operators are used to do an https://en.wikipedia.org/wiki/Arithmetic_shift[arithmetic shift] to the bits of an https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/integral-numeric-types[integral numeric] value, either to the left or the right. 
 
-Shifting an integral number by a value greater than their count of bits minus one is equivalent to shifting by the value https://en.wikipedia.org/wiki/Modulo[modulo] the bit count of the number.
+[source,csharp]
+----
+var number = 7;        // ...00111 (7)
+var one = number << 1; // ...01110 (14) 
+var two = number << 2; // ...11100 (28)
```
[the remark this answers is not in the document]
Comment (gregory-paidis-sonarsource) [reply rules/S2183/csharp/rule.adoc:11]: I really like this suggestion!

**Pull request merged**